src/Overlay.tsx

```diff
@@ -0,0 +1,90 @@
+import styled from 'styled-components'
+import React, {ReactElement} from 'react'
+import {get, COMMON, POSITION} from './constants'
+import {ComponentProps} from './utils/types'
+import { useOverlay, AnchoredPositionHookSettings, TouchOrMouseEvent } from './hooks'
+import Portal from './Portal'
+
+type StyledOverlayProps = {
+  width?: keyof typeof widthMap
+  height?: keyof typeof heightMap
+  visibility: 'visible' | 'hidden'
+}
+
+const heightMap = {
+  'sm': '480px',
+  'md': '640px',
+  'auto': 'auto'
+}
+
+const widthMap = {
+  'sm': '256px',
+  'md': '320px',
+  'lg': '480px',
+  'xl': '640px',
+  'auto': 'auto'
+}
+
+/*TODO replace with shadow functional color variable when it's shipped to primer/primitives*/
+const StyledOverlay  = styled.div<StyledOverlayProps>`
+  background-color: ${get('colors.bg.overlay')};
+  box-shadow: ${get('overlay.boxShadow')};
+  position: absolute;
+  min-width: 192px;
+  max-width: 480px;
+  height: ${props => heightMap[props.height || 'auto']};
+  width: ${props => widthMap[props.width || 'auto']};
+  border-radius: ${get('overlay.borderRadius')};
+  overflow: hidden;
+  animation: overlay-appear 200ms ${get('animation.easeOutCubic')};
+
+  @keyframes overlay-appear {
+    0% {
+      opacity: 0;
+      transform: translateY(${get('space.2')});
```

Comment:
_Sorry I didn't flesh all the details for this transition in the prototype or the docs! I'm including some details here but feel free to work on these changes in a different PR!_

The X/Y translation when an overlay appears should be tied to the anchor side. That would be specific to the `outside-*` options:

`outside-top`: the overlay slides _to the top_, so it feels like it was coming from the `source`.
`outside-bottom`: slides to the bottom (see example below)
`outside-left` slides to the left
`outside-right` slides to the right

https://user-images.githubusercontent.com/293280/111356276-72b10000-8645-11eb-91a0-0864235edff7.mov